In [3]:
import numpy as np
import scipy.stats as stats

#### __Задача 1__

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

Во-первых, у нас известно мат. ожидание и дисперсия, а значит, известно и какую статистику будем применять.

H0: мат. ожидание диаметра подшипников равно 17 мм (а = 17).

n = 100 - количество элементов в выборке

D = 4 мм -> сигма = 2 мм

Мат. ожидание выборки 17.5 мм

Функция статистики примет вид:

$$T(X) = \dfrac{\overline{X} - 17}{2 / 10} = 5 \cdot (\overline{X} - 17).$$

Уровень значимости альфа = 0.025. Найдем квантиль для правосторонней критической области:

In [4]:
n = 100
a = 17
sigma = 2
alpha = 0.025

q = stats.norm.ppf(alpha / 2)
q


-2.241402727604945

In [6]:
#Сгенерируем выборку 100 подшипников с a=17.5 и sigma=2
n = 100
samples = np.random.normal(17.5, 2, size=n)
print(samples)

[16.96020355 20.13497158 18.1654137  17.77906001 13.66187457 13.46489674
 17.36400054 14.18384343 16.29905412 18.06152129 20.38461006 18.29428947
 19.02120134 15.92933462 19.17386459 20.7715087  16.99518538 14.51019618
 16.12791937 18.73773014 21.38500777 18.21305759 21.14161619 17.26444932
 18.3119433  17.42249559 18.05492112 15.29016829 14.24899446 16.86595956
 16.37733032 19.40219368 20.73250524 16.07610739 18.77700604 17.58465214
 16.01715453 15.6320824  18.94593798 17.68574624 20.75821487 20.43579276
 18.14513533 14.80012249 18.33524755 17.85210633 16.63698567 20.34732784
 19.18964712 18.28412277 16.91646746 19.52154356 17.42513859 13.46673138
 19.86685952 16.80991341 15.13510005 14.91215376 18.00304891 20.39530532
 18.328818   20.10903056 17.50725459 15.62994449 17.93143499 19.77985532
 15.18612514 16.08713838 13.86183205 18.99740336 19.51957999 21.18653893
 15.98627369 18.03970069 17.29232206 16.11332858 17.08553432 21.74526706
 19.35330435 16.22043627 18.8127192  20.25142211 22

In [7]:
meanT = samples.mean()
meanT

17.76828538797346

In [9]:
#Проверим гипотезу, посчитав статистику:
T = 5*(meanT-17)
T

3.8414269398673007

Как видим, значение не попало в критическую область (вышло за ее пределы), так что гипотеза о среднем диаметре подшипников в 17 мм неверна.

__Задача 2__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания `a` с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

$$
P \left(
    \overline{X} - t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
    \leq a \leq 
    \overline{X} + t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
\right) = p.
$$

Если p = 0.95, то альфа = 0.05

In [21]:
sigma = 16
p = 0.95
alpha = 0.05
X = 80
n = 256
df = n-1

In [22]:
q2 = stats.norm.ppf((1 - alpha) / 2)
q2

-0.06270677794321385

In [23]:
# Интервал
left = X - q2*sigma/np.sqrt(n)
right = X + q2*sigma/np.sqrt(n)
print(f'{left} <= a <= {right} = {p}')

80.06270677794322 <= a <= 79.93729322205678 = 0.95


__Задача 3__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

H0 гипотеза: средний вес пачки = 200 г (т.е. мат. ожидание = 200). 

alpha = 0.01

n = 10

Распределение нормальное, но больше ничего неизвестно.


In [28]:
a = 200
alpha = 0.01
n = 10
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

Найдём среднее и несмещённую оценку для среднего квадратического отклонения

In [29]:
mean = samples.mean()
std = samples.std(ddof=1)
print(f'mean: {mean}, std: {std}')

mean: 198.5, std: 4.453463071962462


In [30]:
# Найдем квантиль
q3 = stats.t.ppf(1 - alpha / 2, df=n - 1)
q3

3.2498355440153697

Теперь нужно посчитать t-статистику Стьюдента:
$$T(X) = \dfrac{\overline{X} - a}{\sigma_X / \sqrt{n}},$$
$$T(X) = \dfrac{\overline{mean} - a}{std / \sqrt{n}},$$

In [31]:
T = (mean-a)/std*np.sqrt(n)
T

-1.0651074037450896

Так как по данной статистике интервал критической области значимости будет от -бесконечности до -3.2498 и от 3.2498 до + бесконечности, а Т=-1.0651, то можем заключить, что гипотеза верна и мат. ожидание может быть равно 200.